### ISSUE TITLE
Created a block with no assigned slots... Is my CNCLI playing up ?

### ISSUE THREAD

**Issue by Mistralteam** [AndrewWestberg/cncli#80]
Hi All, 

So I have only been running nodes for the last 5 weeks.. Every 36 hours prior, I run CNCLI and even when specifying "NEXT" It shows the current epoch, 

Never the less, this epoch I had run CNCLI for "Current" and discovered i was not going to create a block,

I check my server every morning and Noticed i minted a block over night. 

Is this normal? Should I have created a block even though I had none assigned? 

Thanks.

**Comment by AndrewWestberg** [comment]
Your setup must be misconfigured. CNCLI is perfectly accurate in predicting blocks. Please post the script you're running to check the next epoch so I can help.

**Comment by rafek47** [comment]
Hello Andrew, 
I've found I have the same issue and it seems I am not alone for sure. Noticed another SPOs having the same problem.

I am using cntools / cncli installed with prereqs.sh -c
Node version is 1.27.0
In one of the node updates (1.26 or .27 - not sure) cncli stopped working so I needed to reinstall it via prereqs.sh -c -f
When I checked blocks for the epoch 270 it showed me 0 slots, but I was surprised by a block in that epoch. 
Since it was minted it showed up in my gLiveview as adopted with the slot number. 
--------
This epoch I had the opposite problem, leaderlog showed me an assigned slot, but my pool was not a slotleader in that slot. 
Both cntools and running script manually showed me:
  "no": 1,
      "slot": 32648820,
      "slotInEpoch": 76020,
      "at": "2021-06-20T20:51:51+02:00"

and by logs showing: 
:{"thread":"614","loc":null,"data":{"val":{"kind":"TraceNodeNotLeader","slot":32648820},"  << not leader here, not missed

Finally it seems my current's epoch nonce is not correct (comparing with another SPO) 
"epochNonce": "e2b30667b885672ca223f23dcf266dea9354e4492afddfa92effaab3bbbd0b64", << mine

Did some research and know at least 2 other SPOs with the same problem, showing incorrect slots and minting in another ones.
And found this  https://forum.cardano.org/t/missing-blocks-but-unsure-why/52266/20  << same issue, incorrect epoch nonce

**Comment by AndrewWestberg** [comment]
Db is probably corrupted. I've seen it happen if it was pointed at different networks. I would delete cncli.db and then restart.

**Comment by rafek47** [comment]
Hi thank you so much. I did the following 
1. deleted cncli.db
2. restarted cnode-cncli-sync.service
and cnode-cncli-leaderlog.service
3. waited for db sync
4. ran cncli leaderlog force (cntools) 

If I run the script manually, it shows the correct nonce now (ep 273): ed84a7351c2e893dae79ce2ef6c1e628d46dee63dc548a78fdba88562c62e41b 

incorrect slots are still showing up in gLiveview and cntools, but some new slotleader was added for the next epoch.

Thanks a lot for your help.
